File: sdk/SDK.Examples/src/GetSigningStatusExample.cs
using System;
using System.IO;
using Silanis.ESL.SDK;
using Silanis.ESL.SDK.Builder;

namespace SDK.Examples
{
	public class GetSigningStatusExample : SDKSample
	{
        public static void Main (string[] args)
        {
            new GetSigningStatusExample().Run();
        }

        public SigningStatus draftSigningStatus, sentSigningStatus, trashedSigningStatus;

        override public void Execute()
        {
            DocumentPackage package = PackageBuilder.NewPackageNamed (PackageName)
					.DescribedAs ("This is a new package")
					.WithSigner(SignerBuilder.NewSignerWithEmail(email1)
					            .WithFirstName("John")
					            .WithLastName("Smith"))
					.WithDocument(DocumentBuilder.NewDocumentNamed("My Document")
                                  .FromStream(fileStream1, DocumentType.PDF)
					              .WithSignature(SignatureBuilder.SignatureFor(email1)
					              		.OnPage(0)
					               		.AtPosition(500, 100)))
					.Build ();

            PackageId packageId = eslClient.CreatePackage (package);
            draftSigningStatus = eslClient.GetSigningStatus(packageId, null, null);

            eslClient.SendPackage(packageId);
            sentSigningStatus = eslClient.GetSigningStatus(packageId, null, null);

            eslClient.PackageService.Trash(packageId);
            trashedSigningStatus = eslClient.GetSigningStatus(packageId, null, null);
		}
	}
}

File: sdk/SDK.VBExamples/SDK.VBExamples/GetSigningStatusExample.vb
Imports System
Imports System.IO
Imports Silanis.ESL.SDK
Imports Silanis.ESL.SDK.Builder

Namespace SDK.VBExamples
    Public Module GetSigningStatusExample
        Dim apiToken = "YOUR TOKEN HERE"
        Dim baseUrl = "ENVIRONMENT URL HERE"

        Sub Main()
            ' Create new esl client with api token and base url
            Dim client = New EslClient(apiToken, baseUrl)
            Dim file = New FileInfo(Directory.GetCurrentDirectory() + My.Resources.document)

            Dim package = PackageBuilder.NewPackageNamed("VB Package " + DateTime.Now) _
                .DescribedAs("This is a new package") _
                .WithSigner(SignerBuilder.NewSignerWithEmail("john.smith@email.com") _
                    .WithFirstName("John") _
                    .WithLastName("Smith")) _
                .WithDocument(DocumentBuilder.NewDocumentNamed("My Document") _
                    .FromFile(file.FullName) _
                    .WithSignature(SignatureBuilder.SignatureFor("john.smith@email.com") _
                        .OnPage(0) _
                        .AtPosition(500, 100))) _
                .Build()

            Dim id = client.CreatePackage(package)

            Dim status = client.GetSigningStatus(id, Nothing, Nothing)
            Console.WriteLine("Status after creation = " + status)

            client.SendPackage(id)

            Console.WriteLine("Package {0} was sent", id.Id)
            status = client.GetSigningStatus(id, Nothing, Nothing)

            Console.WriteLine("Status after sending out package = " + status)
        End Sub
    End Module
End Namespace